smart-contracts/contracts/UnlockTestV2.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity 0.4.24;
+
+/**
+ * @title Example upgrade of the Unlock contract
+ * @author Julien Genestoux (unlock-protocol.com)
+ * @author HardlyDifficult
+ * 
+ * This inherits from the previous version, adds data and modifies logic.
+ */
+
+import "./Unlock.sol";
+
+
+contract UnlockTestV2 is Unlock {
+
+  // Example new data (which must come after the original data)
+  bool internal initializedV2;
+  uint public exampleData;
+
+  // Adding a second initialize for the new data as 'initialized' is already true when v2 is deployed.
+  function initializeV2()
+    public 
+  {
+    require(!initializedV2);
```

Comment:
Is that mechanism required? (adding a check on whether the current version has been deployed)

Reply:
Also, which part is invoking `initializeV2` ?

Reply:
We need to ensure in it only happens once.

You can do it with the same name for each contract version, but it [confuses Truffle](https://docs.zeppelinos.org/docs/advanced.html). That doc has an alternative work around we could use.

The test shows calling on upgrade.

Reply:
Cool. Thanks!